startup_scripts/210_vlans.py

```diff
@@ -0,0 +1,57 @@
+from dcim.models import Site
+from ipam.models import VLAN, VLANGroup, Role
+from ipam.constants import VLAN_STATUS_CHOICES
+from tenancy.models import Tenant, TenantGroup
+from extras.models import CustomField, CustomFieldValue
+from ruamel.yaml import YAML
+
+from pathlib import Path
+import sys
+
+file = Path('/opt/netbox/initializers/vlans.yml')
+if not file.is_file():
+  sys.exit()
+
+with file.open('r') as stream:
+  yaml = YAML(typ='safe')
+  vlans = yaml.load(stream)
+
+  optional_assocs = {
+    'site': (Site, 'name'),
+    'tenant': (Tenant, 'name'),
+    'tenant_group': (TenantGroup, 'name'),
+    'group': (VLANGroup, 'name'),
+    'role': (Role, 'name')
+  }
+
+  if vlans is not None:
+    for params in vlans:
+      custom_fields = params.pop('custom_fields', None)
+
+      for assoc, details in optional_assocs.items():
+        if assoc in params:
+          model, field = details
+          query = { field: params.pop(assoc) }
+
+          params[assoc] = model.objects.get(**query)
+
+      if 'status' in params:
+        for vlan_status in VLAN_STATUS_CHOICES:
+          if params['status'] in vlan_status:
+            params['status'] = vlan_status[0]
```

Comment:
Sure, I didn't add it because in the example you sent it wasn't done https://github.com/netbox-community/netbox-docker/blob/master/startup_scripts/110_devices.py#L52-L55 , since those lists are like 3 items each I thought you didn't want to use breaks. Now it's there.

Reply:
I guess that's an error in that script as well ;) Would you mind fixing that one as well?